jupyter_server/services/contents/filemanager.py

```diff
@@ -433,16 +433,15 @@ def save(self, model, path=""):
         """Save the file model and return the model with no content."""
         path = path.strip("/")
 
+        os_path = self._get_os_path(path)
+        self.log.debug("Saving %s", os_path)
```

Comment:
The log statement should probably still be kept where it was.

Reply:
Previously we had the log statement before the hook, I kept that order. Any reason to change it?

Reply:
I mainly meant that in the error condition, we will log "Saving /mypath/foo", and then error out due to misformed content/metadata. I was mainly thinking that stating " I will save a file now" should be done as close as possible to the actual save call. So yes, just a nitpick.